client/lib/wp/sync-handler/cache-index.js

```diff
@@ -84,13 +88,13 @@ export const cacheIndex = {
 
 	dropOlderThan( lifetime ) {
 		const dropElders = records => {
-			const droppedRecords = filter( records, rec => {
+			const removeRecords = filter( records, rec => {
```

Comment:
Not following your meaning here. How are you suggesting we do this in a single pass?